src/files/file_path.h

```diff
@@ -192,13 +190,13 @@ class FilePath {
   // only contains one component, returns a FilePath identifying
   // kCurrentDirectory.  If this object already refers to the root directory,
   // returns a FilePath identifying the root directory.
-  FilePath DirName() const WARN_UNUSED_RESULT;
+  FilePath DirName() const __attribute__((warn_unused_result));
```

Comment:
Hmm, considering `__attribute__` is not standard C++, it might be better to keep this as a macro that's defined to empty on non-GCC compatible compilers. But a better option would be to use C++11 generalized attributes -- these will be ignored by compliant C++11 compilers if they don't support the attribute:

```
FilePath DirName [[gnu::warn_unused_result]] () const;
```